packages/editor/src/components/browser/FunctionBrowser.tsx

```diff
@@ -1,20 +1,153 @@
+          `${row.name.split('.').pop()}${
+            row.isField === false ? `(${row.params.map(param => param.type.split('.').pop()).join(', ')})` : ''
+          }`,
+        id: 'name',
+        cell: cell => {
+          return (
+            <ExpandableCell
+              cell={cell}
+              title={cell.row.original.name}
+              icon={cell.row.original.isField ? IvyIcons.FolderOpen : IvyIcons.Function}
+              additionalInfo={cell.row.original.returnType ? cell.row.original.returnType.simpleName : 'no return type defined'}
+            />
+          );
+        }
+      }
+    ],
+    []
+  );
+
+  const [expanded, setExpanded] = useState<ExpandedState>(false as ExpandedState);
+  const [globalFilter, setGlobalFilter] = useState('');
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+
+  const table = useReactTable({
+    data: tree,
+    columns: columns,
+    state: {
+      expanded,
+      globalFilter,
+      rowSelection
+    },
+    filterFromLeafRows: true,
+    enableRowSelection: true,
+    enableMultiRowSelection: false,
+    enableSubRowSelection: false,
+    onExpandedChange: setExpanded,
+    onGlobalFilterChange: setGlobalFilter,
+    onRowSelectionChange: setRowSelection,
+    getSubRows: row => (row.returnType ? row.returnType.functions : []),
+    getCoreRowModel: getCoreRowModel(),
+    getExpandedRowModel: getExpandedRowModel(),
+    getFilteredRowModel: getFilteredRowModel()
+  });
+
+  useEffect(() => {
+    if (Object.keys(rowSelection).length !== 1) {
+      setShowHelper(false);
+      props.onChange('');
+      return;
+    }
+    const selectedRow = table.getRowModel().rowsById[Object.keys(rowSelection)[0]];
+    const getParentNames = (currentRow: Row<Function>, parentNames: string[] = []): string[] => {
+      parentNames.push(currentRow.original.name);
+      const parentRow = currentRow.getParentRow();
+      if (parentRow !== undefined) {
+        return getParentNames(parentRow, parentNames);
+      }
+      return parentNames;
+    };
+
+    const parentNames = getParentNames(selectedRow);
+    const selectedName = parentNames.reverse().join('.');
+
+    const selectedFunction = selectedRow.original.isField
+      ? selectedName
+      : selectedName + '(' + selectedRow.original.params.map(param => param.type.split('.').pop()).join(', ') + ')';
+    const paramTypesArray: string[] = selectedRow.original.params.map(param => param.type);
+    const parentRow = selectedRow.getParentRow();
+    const type = parentRow
+      ? parentRow.original.returnType.packageName + '.' + parentRow.original.returnType.simpleName
+      : selectedRow.original.returnType.packageName + '.' + selectedRow.original.returnType.simpleName;
+    setType(type);
+    setMethod(selectedRow.original.name);
+    setParamTypes(paramTypesArray);
+    setSelectedFunctionDoc(doc);
+    setShowHelper(true);
+    props.onChange(selectedFunction);
+  }, [doc, props, rowSelection, table]);

```

Comment:
I have no idea what is happening here...
Any maybe you could gain also more speed / performance if you can split this into smaller parts...
This use effect is now triggered if the doc, the props, a rowSelection or the whole table changes...

Reply:
Yes, you are right, it's a bit confusing and the useEffect currently reacts to many changes. However, I wouldn't know how to split it up better because, when selecting a row, I need to perform all these tasks. 

In general, this useEffect is meant to control the behavior when a row is selected. When a row is selected, I need to first traverse back through the tree (using the getParentNames method) to obtain the complete name of the method. All the subsequent set... operations are required for calling the JavaDoc/helpertext. I've now tried to structure the code a bit better. If it's okay with you, I would merge the code like this and in a new PR, attempt to improve it (if possible). I'll have to modify the code again anyway to optimize the search.
![grafik](https://github.com/axonivy/inscription-client/assets/141223521/b1cafcfb-27b7-4765-b3e9-07e875170a12)

Reply:
Yes I know that this is used for the row selection, but it runs also if the tree changes, and the tree changes e.g. when you search for something. 
But yes, merge it first then we can improve it :)